### PR TITLE
Use aws_s3_object instead of aws_s3_bucket_object

### LINKED ISSUE #801: update Terraform code to prevent Deprecated Resource warnings
When deploying TEA, I get the following Terraform deprecation warning:

```
Warning: Deprecated Resource

  with module.thin_egress_app.aws_s3_bucket_object.lambda_source,
  on .terraform/modules/thin_egress_app/main.tf line 25, in resource "aws_s3_bucket_object" "lambda_source":
  25: resource "aws_s3_bucket_object" "lambda_source" {

use the aws_s3_object resource instead
```

### PR DESCRIPTION
**Pull request by gjclark**
Closes #801

### CODE REVIEW

**Review by mckadesorensen — APPROVED** [review]

**Review by mattp0 — APPROVED** [review]

**Review by reweeden — COMMENTED** [review]
I wonder if it would be a good idea to add a `required_providers` block to the terraform module so we can pin the minimum supported version of the AWS provider. I'm actually a little surprised that it works without that.

**Comment by gjclark** [comment]
> I wonder if it would be a good idea to add a `required_providers` block to the terraform module so we can pin the minimum supported version of the AWS provider. I'm actually a little surprised that it works without that.

NISAR has:
```
terraform {
  required_providers {
    aws = {
      source  = "hashicorp/aws"
      version = "~> 5.0"
    }
    null = {
      source  = "hashicorp/null"
      version = "~> 2.1"
    }
  }
}
```
Should I just duplicate that here?

**Comment by reweeden** [comment]
You should only include the providers that the module actually needs and you should use `>=` as explained on the best practices section here: https://developer.hashicorp.com/terraform/language/providers/requirements#best-practices-for-provider-versions

File: .gitignore [hunk 1]
```diff
@@ -11,7 +11,6 @@
 
 # Tools
 .terraform
-/terraform/providers.tf
```
Comment (reweeden) [inline .gitignore:14]: Lol, why was that in there?

File: terraform/providers.tf [hunk 2]
```diff
@@ -0,0 +1,8 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 5.0"
+    }
+  }
+}
```
Comment (reweeden) [inline terraform/providers.tf:8]: Missing newline

File: terraform/providers.tf [hunk 3]
```diff
@@ -0,0 +1,8 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 5.0"
```
Comment (reweeden) [inline terraform/providers.tf:5]: This isn't the minimum. I just checked a random 4.x version and 4.63 has the `aws_s3_object` already. So you can go down at least to that.

You should find the lowest version that has that resource. I would do it through trial and error using this page and selecting the version from the version dropdown:
https://registry.terraform.io/providers/hashicorp/aws/4.63.0/docs/resources/s3_bucket_object

File: .gitignore (hunk 1 above)
Comment (gjclark) [reply .gitignore:14]: Hah I don't know. I was hoping to find some discussion in the PR that brought that in. Couldn't find the reason.

File: terraform/providers.tf (hunk 3 above)
Comment (gjclark) [reply terraform/providers.tf]: Looks like 4.0 introduced it.

**Review by reweeden — APPROVED** [review]

**Pull request merged**